src/FakeItEasy.ruleset

```diff
@@ -1,9 +1,11 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
 <RuleSet Name="FakeItEasy" ToolsVersion="11.0">
   <Include Path="allrules.ruleset" Action="Default" />
   <Rules AnalyzerId="Microsoft.Analyzers.ManagedCodeAnalysis" RuleNamespace="Microsoft.Rules.Managed">
     <Rule Id="CA1020" Action="None" />
+    <Rule Id="CA1200" Action="None" />
     <Rule Id="CA1305" Action="None" />
+    <Rule Id="CA2235" Action="None" />

```

Comment:
"Mark all non-serializable fields". I think occurrences of this warning are false positive, because they only appears in the netstandard1.6 build, where serialization is disabled... It seems to be ignoring the `#if FEATURE_BINARY_SERIALIZATION`.

Reply:
In that case, should we consider disabling the warning on a case-by-case basis? And/or report the missed `#if`?

Reply:
How would you do this? By putting `#pragma warning disable` between `#if`s? It would look a bit ugly, but it should work.
What I don't understand is that for netstandard1.6, the `[Serializable]` attribute is suppressed, so this warning shouldn't kick in... I'll open an issue in the roslyn-analyzers repo.

Reply:
OK, I investigated a bit more. In fact, the warning is *not* triggered when building netstandard1.6, but in the VS error list, the reported project is "FakeItEasy (netstandard1.6)":

![image](https://user-images.githubusercontent.com/1864281/36090091-2fcabafe-0fdf-11e8-853a-5c9ed5ba9eca.png)

Intellisense and command line build report it only in net40/net45. So it looks like a bug in VS.

So, maybe we shouldn't disable this warning entirely, but just suppress it where appropriate. Anyway, I think this warning is a good indication that the serialization of fakes is flaky at best...

Reply:
> So, maybe we shouldn't disable this warning entirely, but just suppress it where appropriate. Anyway, I think this warning is a good indication that the serialization of fakes is flaky at best...

Or maybe not... there's clearly something wrong with the analyzer: the types it reports as not serializable *are* actually serializable (e.g. object, MethodInfo...).

Reply:
Ah! https://github.com/dotnet/roslyn-analyzers/issues/1510

Reply:
Thanks for the pragmas and pointer to the roslyn-analyzer issue.